features/FEATURE_LWIP/lwip-interface/lwip-sys/lwip_random.c

```diff
@@ -18,11 +18,28 @@
 #include "lwip/debug.h"
 #include "lwip/def.h"
 #include "lwip_random.h"
+
+#if defined(DEVICE_TRNG)
+#include "hal/trng_api.h"
+#endif
+
 #include "randLIB.h"
 
 void lwip_seed_random(void)
 {
+#if defined(DEVICE_TRNG)
```

Comment:
What's going on here? LWIP is not using `rand()`, it's using randLIB, so this defeats lwIP's seeding.

If your driver needs `rand()` seeded, that's your issue, not anything to do with lwIP. And this won't achieve anything if you're not using lwIP.

If you really need to do this, it should be in your driver. However, conceptually I like to think `rand()` is for applications, so applications should be free to manually seed it to something non-random, without being bothered by drivers using it or seeding it in the background.

I would suggest you also use randLIB as your RNG source - that's doesn't function as a a manually-seedable PRNG, so you can't interfere with other users.

Reply:
There are two cases in this file either FEATURE_COMMON_PAL is defined or not. randLIB is only used when FEATURE_COMMON_PAL is defined if you look at the whole file. Shouldn't the seed behavior be the same for both?
We don't use rand in the driver but lwIP uses it for port assignment and depending on if FEATURE_COMMON_PAL is defined or not the behavior(seed or no seed) differs I guess?

We can remove the seeding it if it's more suited somewhere else.

Reply:
Look again - there aren't two cases in this file any more; `FEATURE_COMMON_PAL` was eliminated a few weeks back, so randLIB is always in use and the 'srand' isn't achieving anything. Presumably this is the result of a rebase.

Anyway, there's no time do be doing any more general non-EMAC-related lwIP fixes before 5.9 release - only things that are blocking actual EMAC integration like #6926

Reply:
Ok, thanks. That explains it. We'll remove it.